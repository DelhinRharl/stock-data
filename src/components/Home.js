import { NavLink } from 'react-router-dom';

const Home = (item) => {
  const {
    ticker, price, changes, companyName,
  } = item;
  return (

    <div className="">
      <NavLink to={{ pathname: `/details/${ticker}` }}>
        <div className="bg-[#0e128a] bg-gradient-to-r from-[#0e128a] to-[#01034d] w-full h-40 border-b-2 border-blue-800 p-4">
          <div className="text-[#F64C71] text-xl">
            {companyName}
          </div>
          <div className="text-gray-400">
            {ticker}
          </div>
          <div className="text-white text-lg">
            Price:
            {' '}
            {price}
          </div>
          <div className="text-green-500">
            Price Change:
            {' '}
            <span className="text-[#F64C71]">{changes}</span>
          </div>
        </div>
      </NavLink>
    </div>
  );
};

export default Home;
