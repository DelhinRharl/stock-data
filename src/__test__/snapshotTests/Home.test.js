import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import Home from '../../components/Home';
import store from '../../redux/configureStore';

describe('Home  component', () => {
  it(' should render the  component successfully', () => {
    const home = render(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
        ,
      </Provider>,
    );

    expect(home).toMatchSnapshot();
  });
});
