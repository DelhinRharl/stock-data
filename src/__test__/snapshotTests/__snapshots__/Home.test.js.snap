// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home  component  should render the  component successfully 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
      >
        <div
          class="bg-[#0e128a] bg-gradient-to-r from-[#0e128a] to-[#01034d] w-full h-40 border-b-2 border-blue-800"
        >
          <div
            class="text-[#F64C71] text-xl"
          />
          <div
            class="text-gray-400"
          />
          <div
            class="text-white text-lg"
          >
            Price: 
          </div>
          <div
            class="text-green-500"
          >
            Price Change: 
            <span
              class="text-[#F64C71]"
            />
          </div>
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class=""
    >
      <div
        class="bg-[#0e128a] bg-gradient-to-r from-[#0e128a] to-[#01034d] w-full h-40 border-b-2 border-blue-800"
      >
        <div
          class="text-[#F64C71] text-xl"
        />
        <div
          class="text-gray-400"
        />
        <div
          class="text-white text-lg"
        >
          Price: 
        </div>
        <div
          class="text-green-500"
        >
          Price Change: 
          <span
            class="text-[#F64C71]"
          />
        </div>
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
